test/ToolsTestFileIO.cpp

```diff
@@ -954,6 +935,56 @@ TEST_F(TestFileIO, DISABLED_System_Performance_FileIO_DirectoryReader__vs_Boost_
    timeCheck = timeToFind.ElapsedMs();
    std::cout  << "Boost filesystem found           " << boostFileCounter << " items in: " 
        << timeCheck << " millisec" << std::endl;
+}
 
+std::string TestFileIO::CreateTestDirectoryAndFiles(const std::vector<std::string>& filenamesToTouch, const std::string& newDirName = "TestDir") {
+   // Create directory
+   std::string newDirectoryPath = mTestDirectory + std::string("/") + newDirName;
+   std::string createdDirectoryPath = CreateSubDirectory(newDirName, mTestDirectory);
+   EXPECT_TRUE(FileIO::DoesDirectoryExist(newDirectoryPath));
+   EXPECT_EQ(newDirectoryPath, createdDirectoryPath);
 
+   // Add files to directory
+   for (const auto& filename : filenamesToTouch) {
+      std::string createdFilePath = CreateFile(createdDirectoryPath, filename);
+      EXPECT_TRUE(FileIO::DoesFileExist(createdFilePath));
+   }
+   return createdDirectoryPath;
+}
```

Comment:
Can you add a test to have hidden files and see if they are added to the directory contents?